src/components/views/Header.js

```diff
@@ -2,39 +2,77 @@ import React from 'react';
-                                </ul>
-                            </li>
-                        );
-                    }
-                })}
-            </ul>
-        </nav>
-    </header>
-);
+export default class extends React.Component {
+    constructor(props) {
+        super(props);
+
+        // These dashboards do not support custom regions
+        this.regionlessDashboards = [
+            'hardware',
+        ];
+
+        // The preferred region is the region that is being applied to the
+        // current dashboard. If the current dashboard does not support regions
+        // or if a dashboard is not currently loaded, the user's preferred
+        // region might have been recorded in session storage instead.
+        this.state = {
+            preferredRegion: URLManager.getQueryParameter('region') || sessionStorage.getItem('region'),
+        };
+
+        // Update state, which in turns re-renders header links, whenever the
+        // region changes
+        const root = document.getElementById('root');
+        root.addEventListener('regionChanged', () => {
+            this.setState({
+                preferredRegion: URLManager.getQueryParameter('region') || sessionStorage.getItem('region'),
+            });
+        }, false);
+    }
+
+    render() {
+        return (
+            <header id="main-header">
+                <h1><Link to="/">Firefox Public Data Report</Link></h1>
+                <nav id="main-navigation">
+                    <ul>
+                        {dashboards.sections.map(dashboardSection => {
+                            if (dashboardSection.comingSoon) {
+                                return (
+                                    <li key={dashboardSection.key} id={`navigation-${dashboardSection.key}`} className="navigation-coming-soon">
+                                        <span className="navigation-section-title">{dashboardSection.title}</span>
+                                        <span className="navigation-coming-soon-message">Coming soon</span>
+                                    </li>
+                                );
+                            } else {
+                                return (
+                                    <li key={dashboardSection.key} id={`navigation-${dashboardSection.key}`} className="navigation-section">
+                                        <span className="navigation-section-title">{dashboardSection.title}</span>
+                                        <ul className="navigation-section-members">
+                                            {dashboardSection.members.map(dashboard => {
+                                                let linkURL = `/dashboard/${dashboard.key}`;
+
+                                                if (this.state.preferredRegion && !this.regionlessDashboards.includes(dashboard.key)) {
+                                                    linkURL += `?region=${this.state.preferredRegion}`;
```

Comment:
is `region` the only URL param? Might need more logic here to append via `&` otherwise.

Reply:
Good call. It's the only URL param right now, but there could be more in the future. I'll handle that case.

Reply:
Updated. Great catch. Thank you.